src/clj/harja/kyselyt/kanavat/kohteet.clj

```diff
@@ -89,17 +91,58 @@
                                         {::m/poistettu? false})})
     (partial hae-kohteiden-urakkatiedot db user))))
 
-(defn hae-urakan-kohteet [db user urakka-id]
-  (->>
-    (sort-by :harja.domain.kanavat.kohde/jarjestys (specql/fetch db
-                  ::kohde/kohde
-                  (set/union
-                    kohde/perustiedot
-                    kohde/kohteen-kohdekokonaisuus
-                    kohde/kohteenosat)
-                  {::m/poistettu? false}))
-    (hae-kohteiden-urakkatiedot db user urakka-id)
-    (remove (comp empty? ::kohde/urakat))))
+(defn lue-postgres-vector [s]
+  (when (seq s)
+    (str/split s #"\s+")))
+
+(defn lue-postgres-object 
+  [^org.postgresql.util.PGobject x]
+  (when-let [val (.getValue x)]
+    (mapv read-string (lue-postgres-vector val))))
+
+(defn hae-urakan-kohteet 
+  "Kohdekokonaisuus sekä kohteenosat pitää järjestää etelästä pohjoiseen ryhmitettynä"
```

Comment:
Voiko kohdekokonaisuuden kulkusuuntaa tietää etukäteen jostakin?
Tietoa voisi käyttää sitten sorttauksessa, jolloin kohdekokonaisuuden kohteet tiedon perusteella sortataan joko etelä-pohjoinen tai länsi-itä ym. suuntaisesti.

Reply:
En tiedä voiko tuota mitenkään järkevästi koodiin kirjoittaa. Kulkusuunta on sinne missä seuraava kohde (kohdekokonaisuudessa) sijaitsee kun kuljetaan vesiväylää pitkin. En usko että meillä mitään funktiota mikä voisi tietää minne suuntaan vesiväylä kartalla menee.

Taulun kan_kohde id:n avulla voisi tietää mikä kohde on seuraavana, jos id:t olisi järjestyksessä mutta tarkistin juuri tietokannasta menekö id:t kulkusuunnan mukaisesti, ei mene. Joten tähän pitäisi tehdä jokin uusi toiminnallisuus, ei aleta ainakaan tähän haaraan tekemään. Tuo etelä pohjoinen oli kuulemma hyvä.

Asiakkaan kommentit 

_

> Joo tuo etelä-pohjoinen järjestys on ihan hyvä. Se ajaa lähes täysin saman asian kuin väylän nimelliskulkusuunnan mukainen järjestys. Muistaakseni tämä ei ole ollut meillä kyllä missään vaiheessa määrittelyissä mukana
> 
> Eli tässä kanavien ja avattavien siltojen tapauksessa järjestys olisi se mikä sulku- tai siltakohde vesiväylää pitkin ajettaessa seuraavaksi tulee vastaan.
>
> Ja jos tosiaan saa tuon kohdekokonaisuuden vielä otettua mukaan tuohon järjestykseen, niin oikein hyvä!

_